ci/gpu/build.sh

```diff
@@ -8,69 +12,118 @@ function logger() {
   echo -e "\n>>>> $@\n"
 }
 
+# apt-get install libnuma libnuma-dev
+
 # Arg parsing function
 function hasArg {
     (( ${NUMARGS} != 0 )) && (echo " ${ARGS} " | grep -q " $1 ")
 }
 
 # Set path and build parallel level
 export PATH=/conda/bin:/usr/local/cuda/bin:$PATH
-export CUDA_REL=${CUDA//./}
+export PARALLEL_LEVEL=4
+export CUDA_REL=${CUDA_VERSION%.*}
 
 # Set home to the job's workspace
 export HOME=$WORKSPACE
 
-# Switch to project root; also root of repo checkout
-cd $WORKSPACE
-
 # Parse git describe
+cd $WORKSPACE
 export GIT_DESCRIBE_TAG=`git describe --tags`
 export MINOR_VERSION=`echo $GIT_DESCRIBE_TAG | grep -o -E '([0-9]+\.[0-9]+)'`
-
-# Enable NumPy's __array_function__ protocol (needed for NumPy 1.16.x,
-# will possibly be enabled by default starting on 1.17)
-export NUMPY_EXPERIMENTAL_ARRAY_FUNCTION=1
+export UCX_PATH=$CONDA_PREFIX
 
 ################################################################################
 # SETUP - Check environment
 ################################################################################
 
-logger "Get env..."
+logger "Check environment..."
 env
 
+logger "Check GPU usage..."
+nvidia-smi
+
 logger "Activate conda env..."
 source activate gdf
+conda install "cudf>=0.10" "dask-cudf>=0.10" "cudatoolkit=$CUDA_REL" \
+              "dask>=2.3.0" "distributed>=2.3.2" "numpy>=1.16" -c rapidsai-nightly
+
+# needed for asynccontextmanager in py36
+conda install -c conda-forge "async_generator" "automake" "libtool" \
+                              "cmake" "automake" "autoconf" "cython" \
+                              "pytest" "pkg-config" "pytest-asyncio"
+
+
+# install cupy
+conda install -c rapidsai -c nvidia "cupy>=6.2.0"
+
+# install ucx from conda-forge rc channel
+# conda install -c conda-forge/label/rc_ucx "ucx-proc=*=gpu" "ucx"
+
+# Install the master version of dask and distributed
+logger "pip install git+https://github.com/dask/distributed.git --upgrade --no-deps"
+pip install "git+https://github.com/dask/distributed.git" --upgrade --no-deps
+logger "pip install git+https://github.com/dask/dask.git --upgrade --no-deps"
+pip install "git+https://github.com/dask/dask.git" --upgrade --no-deps
 
 logger "Check versions..."
 python --version
-gcc --version
-g++ --version
+$CC --version
+$CXX --version
+conda list
+
+################################################################################
+# BUILD - Build ucx
+################################################################################
+
+logger "Build ucx"
+git clone https://github.com/openucx/ucx
+cd ucx
+ls
+./autogen.sh
+mkdir build
+cd build
+../configure --prefix=$CONDA_PREFIX --enable-debug --with-cuda=$CUDA_HOME --enable-mt CPPFLAGS="-I//$CUDA_HOME/include"
+make -j install
+cd $WORKSPACE
```

Comment:
CI has a time limit, I don't think it's safe to do this. Don't we have conda packages for UCX? If they're outdated it's better to have a new one built.

Reply:
It only takes 1min to build and I think having the newest UCX with debug build flags makes it worth it.